src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_BASE_SIZE = 16;
+    private static final float RESIZE_FACTOR = 0.75f;
+    private static final int RESIZE_COEFFICIENT = 2;
+
+    private Object[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Object[DEFAULT_BASE_SIZE];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+        resize();
+        int position = position(key);
+        if (table[position] == null) {
+            table[position] = new Node<K, V>(key, value, hash(key), null);
+            size++;
+            return;
+        }
+
+        Node<K, V> current = (Node<K, V>) table[position];
+
+        while (current.next != null) {
+            if (areEqual(current.key, key)) {
+                current.value = value;
+                return;
+            }
+            current = current.next;
+        }
+        if (areEqual(current.key, key)) {
+            current.value = value;
+            return;
+        }
+        current.next = new Node<K, V>(key, value, hash(key), null);
```

Comment:
Diamond operator. Use `new Node<>(...)` instead of `new Node<K,V>(...)` to avoid redundancy since Java 7.